### DISCUSSION TITLE
What is vetur.useWorkspaceDependencies for?

### DISCUSSION THREAD

**Discussion by fabis94** [vuejs/vetur#2664] [Q&A] [answered]
I've got a monorepo setup and I followed the guidelines on the Vetur website to set it up. I'm using Vetur in a Nuxt.js TypeScript project and everything seemed to more or less work, except Intellisense/Auto-import suggestions for importing other .ts files.

The moment I removed `vetur.useWorkspaceDependencies` from my config and restarted Vetur, auto import suggestions started appearing.

So what is this setting and why does it break auto-imports? I couldn't find any sort of documentation online properly explaining what this does.

**Comment by yoyo930021** [comment]
When option is true,
Vetur will try to load `dependencies` from your project.
This dependencies include TypeScript package.

TypeScript package provide more features.
Like: JavaScript completion, define.

I think you probably have TypeScript package in your monorepo.
And this package is too old.

You can open output panel in VSCode.
And find message in `Vue language server`.
The message include TypeScript version and TypeScript package path in your project.

**Comment by fabis94** [comment]
> When option is true,
> Vetur will try to load dependencies from your project.
> This dependencies include TypeScript package.

I'm sorry I don't understand what you mean with this. Can you elaborate?

**Comment by yoyo930021** [comment, marked as answer]
Vetur will use TypeScript language service for many features in JavaScript/TypeScript.

TypeScript language server is in TypeScript package.

When this option is true, Vetur will try to find TypeScript package in your project.
If you have custom TypeScript version in your project,
Vetur will use it.

When false, Vetur will use built-in TypeScript package.

This option doesn't only effect TypeScript.
Same behavior for any features needs to other packages.
But your problem look like TypeScript package.

**Comment by fabis94** [comment]
Thank you, I installed `typescript` in monorepo root and this setting started working correctly
